django/db/models/sql/compiler.py

```diff
@@ -830,9 +830,12 @@ def execute_sql(self, result_type=MULTI, chunked_fetch=False):
         try:
             cursor.execute(sql, params)
         except Exception:
-            with self.connection.wrap_database_errors:
-                # Closing a server-side cursor could yield an error
+            try:
+                # Might fail for server-side cursors (eg: connection closed)
                 cursor.close()
+            except Exception:
+                # Ignore clean-up errors. Raise the original error instead.
+                pass
```

Comment:
While we still need to support python 2, how about [https://pythonhosted.org/six/#six.raise_from](six.raise_from) ?